packages/@ember/debug/data-adapter.ts

```diff
@@ -141,7 +141,7 @@ class TypeWatcher {
       consumeTag(tagFor(records, '[]'));
 
       if (hasBeenAccessed === true) {
-        onChange();
+        next(() => onChange());
```

Comment:
Nitpick, but I think you can also do `next(onChange);` and avoid creating a new function?

Reply:
sure, done